src/cmd/dut.rs

```diff
@@ -265,6 +272,287 @@ lazy_static! {
+    }
+    eprintln!("PASS: GBB flags are set for dev");
+    Ok(())
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// Make DUTs connected via Servo ready for development
+/// "Ready for development" means:
+/// - CCD (Closed Case Debugging) is in "Open" state
+/// - A Servo is attached correctly
+/// - At least one Ethernet connection is available (so MAC addr and an IP address is known)
+/// - GBB flags are set to 0x19
+/// - CCD testlab mode is enabled
+#[argh(subcommand, name = "setup")]
+struct ArgsSetup {
+    /// servo serial
+    #[argh(option)]
+    serial: Option<String>,
+    /// cros repo dir to use
+    #[argh(option)]
+    repo: Option<String>,
+    /// do ccd open only
+    #[argh(switch)]
+    open_ccd: bool,
+    /// check ccd status via servo
+    #[argh(switch)]
+    get_ccd_status: bool,
+    /// do gbb flags update only
+    #[argh(switch)]
+    set_dev_gbb_flags: bool,
+    /// reset gbb flags only
+    #[argh(switch)]
+    reset_gbb_flags: bool,
+    /// do ccd unlock only
+    #[argh(switch)]
+    enable_ccd_testlab: bool,
+    /// check if the DUT is reachable via SSH
+    #[argh(switch)]
+    check_ssh: bool,
+}
+fn run_setup(args: &ArgsSetup) -> Result<()> {
+    let repo = get_repo_dir(&args.repo)?;
+    let servo = if let Some(serial) = &args.serial {
+        let list = ServoList::discover()?;
+        list.find_by_serial(serial).context(format!("
+        Servo {serial} not found.
+        Please check the servo connection, try another side of USB port, attach servo directly with a host instead of via hub, etc...
+        `lium servo list` may be helpful.
+        "))?.clone()
+    } else {
+        let list = ServoList::discover()?;
+        let list: Vec<LocalServo> = list
+            .devices()
+            .iter()
+            .filter(|s| s.is_servo())
+            .cloned()
+            .collect();
+        if list.len() != 1 {
+            return Err(anyhow!("Please specify --serial when multiple Servo is connected. `lium servo list` may be helpful."));
+        }
+        list.first()
+            .context(
+                "Servo is not connected. Run `lium servo list` to check if Servo is connected.",
+            )?
+            .clone()
+    };
+    eprintln!("Using {} {} as Servo", servo.product(), servo.serial());
+    let cr50 = get_cr50_attached_to_servo(&servo)?;
+    eprintln!("Using {} {} as Cr50", cr50.product(), cr50.serial());
+    if args.open_ccd {
+        open_ccd(&cr50)?;
+    } else if args.get_ccd_status {
+        get_ccd_status(&cr50)?;
+    } else if args.set_dev_gbb_flags {
+        set_dev_gbb_flags(&repo, &cr50)?;
+    } else if args.reset_gbb_flags {
+        reset_gbb_flags(&repo, &cr50)?;
+    } else if args.enable_ccd_testlab {
+        enable_ccd_testlab(&cr50)?;
+    } else if args.check_ssh {
+        check_ssh(&servo)?;
+    } else {
+        open_ccd(&cr50)?;
+        enable_ccd_testlab(&cr50)?;
+        let dut = check_ssh(&servo)?;
+        check_gbb_flags(&dut)?;
```

Comment:
Is there any reason why this function cannot be run directly with an option? All the other functions called from this else block seem to have an option to run it.

Reply:
Good point! Added a new commit that calls the check_gbb_flags function only. Please review again! (sorry for the delayed response...)